flexmeasures/data/models/charts/belief_charts.py

```diff
@@ -75,38 +83,68 @@ def bar_chart(
 
 
 def chart_for_multiple_sensors(
-    sensors: list["Sensor"],  # noqa F821
+    sensors_to_show: list["Sensor", list["Sensor"]],  # noqa F821
     event_starts_after: datetime | None = None,
     event_ends_before: datetime | None = None,
     **override_chart_specs: dict,
 ):
-    sensors_specs = []
+    # Unpack nested sensors
+    sensors = flatten_unique(sensors_to_show)
+
+    # Determine the shared data resolution
     condition = list(
         sensor.event_resolution
         for sensor in sensors
         if sensor.event_resolution > timedelta(0)
     )
-    minimum_non_zero_resolution_in_ms = (
-        min(condition).total_seconds() * 1000 if any(condition) else 0
-    )
-    for sensor in sensors:
-        unit = sensor.unit if sensor.unit else "a.u."
+    minimum_non_zero_resolution = min(condition) if any(condition) else timedelta(0)
+
+    # Set up field definition for event starts
+    event_start_field_definition = FIELD_DEFINITIONS["event_start"]
+    event_start_field_definition["timeUnit"] = {
+        "unit": "yearmonthdatehoursminutesseconds",
+        "step": minimum_non_zero_resolution.total_seconds(),
+    }
+    if event_starts_after and event_ends_before:
+        event_start_field_definition["scale"] = {
```

Comment:
What happens here? Please provide a comment.

Reply:
It seems you decided against adding a comment?

Reply:
I added a comment in line 106.